src/main/groovy/life/qbic/samplestatus/reporter/services/RealLimsQueryService.groovy

```diff
@@ -0,0 +1,108 @@
+     * <p>Please note, that we need three constructor parameters, that must be configured in the applications property file.</p>
+     *
+     * <p>We have the application property <code>service.openbis.user</code> that maps to the parameter <code>openbisUser</code>,
+     * then the property <code>service.openbis.password</code> that maps to the parameter <code>openbisPassword</code>
+     * and finally the property <code>service.openbis.api-server-url</code> that maps to the parameter <code>applicationServerUrl</code>.</p>
+     *
+     * <p>We also let you configure the server timeout, which uses the application property <code>service.openbis.server-timeout</code>
+     * and maps it to the parameter <code>serverTimeout</code>.
+     *
+     * @param openbisUser the openBIS user id
+     * @param openbisPassword the openBIS user password
+     * @param applicationServerUrl the openBIS application server url
+     * @param serverTimeout the server connection timeout
+     */
+    RealLimsQueryService(@Value('${service.openbis.user}') String openbisUser,
+                         @Value('${service.openbis.password}') String openbisPassword,
+                         @Value('${service.openbis.api-server-url}') String applicationServerUrl,
+                         @Value('${service.openbis.server-timeout}') Integer serverTimeout) {
+        this.openBisApplicationServerApi = HttpInvokerUtils.createServiceStub(
+                IApplicationServerApi.class,
+                applicationServerUrl + IApplicationServerApi.SERVICE_URL, serverTimeout)
+        sessionToken = this.openBisApplicationServerApi.login(openbisUser, openbisPassword)
+    }
+
+    /**
+     * <p>This method is called by the Spring framework, after an instance of this class
+     * has been created.</p>
+     * <p>We just check, if the authentication against openBIS has been successful
+     * which is the case, when there is a session token present.</p>
+     */
+    @PostConstruct
+    void init() {
+        if (!sessionToken) {
+            throw new AuthenticationException("Authentication against LIMS service failed.")
+        }
+    }
+
+    /**
+     * {@InheritDocs}
+     */
+    @Override
+    List<Sample> getUpdatedSamples(Instant updatedSince) {
+        SampleSearchCriteria criteria = new SampleSearchCriteria()
+        // we make sure that the barcode is set, otherwise the sample is of no interest to us
+        criteria.withProperty("QBIC_BARCODE").thatContains("Q")
+        // only fetch latest samples
+        criteria.withModificationDate().thatIsLaterThanOrEqualTo(Date.from(updatedSince))
+ 
+        // we need to fetch properties, as the sample status (and barcode) is contained therein
+        SampleFetchOptions fetchOptions = new SampleFetchOptions()
+        fetchOptions.withProperties()
+ 
+        SearchResult<Sample> result = v3.searchSamples(sessionToken, criteria, fetchOptions)
+ 
+        return result.getObjects()
```

Comment:
Here I return a list of openBIS samples instead of just the barcode wrapper (reporter.Sample). We need to add the LimsMapper here or outside of this function somehow. Also note that the LimsMapper should (imho) return SampleUpdate and not reporter.Sample.